taskt/UI/Forms/frmScriptBuilder.cs

```diff
@@ -1815,413 +1406,575 @@ private void lnkGitWiki_LinkClicked(object sender, LinkLabelLinkClickedEventArgs
-        }
+            try
+            {
+                //deserialize file
+                Script deserializedScript = Script.DeserializeFile(filePath);
 
+                if (deserializedScript.Commands.Count == 0)
+                {
+                    Notify("Error Parsing File: Commands not found!");
+                }
 
-        #endregion
+                //variables for comments
+                var fileName = new FileInfo(filePath).Name;
+                var dateTimeNow = DateTime.Now.ToString();
 
-        #region Automation Engine Delegate
+                //comment
+                lstScriptActions.Items.Add(CreateScriptCommandListViewItem(new AddCodeCommentCommand()
+                {
+                    v_Comment = "Imported From " + fileName + " @ " + dateTimeNow
+                }));
 
-        #endregion
+                //import
+                PopulateExecutionCommands(deserializedScript.Commands);
+                foreach (ScriptVariable var in deserializedScript.Variables)
+                {
+                    if (_scriptVariables.Find(alreadyExists => alreadyExists.VariableName == var.VariableName) == null)
+                    {
+                        _scriptVariables.Add(var);
+                    }
+                }
 
+                //comment
+                lstScriptActions.Items.Add(CreateScriptCommandListViewItem(new AddCodeCommentCommand() { v_Comment = "End Import From " + fileName + " @ " + dateTimeNow }));
 
-        private void button1_Click(object sender, EventArgs e)
-        {
-            UI.Forms.Supplemental.frmThickAppElementRecorder recorder = new Supplemental.frmThickAppElementRecorder();
-            recorder.ShowDialog();
+                //format listview
+                //notify
+                Notify("Script Imported Successfully!");
+            }
+            catch (Exception ex)
+            {
+                //signal an error has happened
+                Notify("An Error Occured: " + ex.Message);
+            }
         }
 
-        private void moveToParentToolStripMenuItem_Click(object sender, EventArgs e)
+        public void PopulateExecutionCommands(List<ScriptAction> commandDetails)
         {
-
-            //create command list
-            var commandList = new List<Core.Automation.Commands.ScriptCommand>();
-
-           //loop each
-            for (int i = lstScriptActions.SelectedItems.Count - 1; i >= 0; i--)
+            foreach (ScriptAction item in commandDetails)
             {
-                //add to list and remove existing
-                commandList.Add((Core.Automation.Commands.ScriptCommand)lstScriptActions.SelectedItems[i].Tag);
-                lstScriptActions.Items.Remove(lstScriptActions.SelectedItems[i]);
+                lstScriptActions.Items.Add(CreateScriptCommandListViewItem(item.ScriptCommand));
+                if (item.AdditionalScriptCommands.Count > 0) PopulateExecutionCommands(item.AdditionalScriptCommands);
```

Comment:
We should try to avoid single line statements like this. Normally I wouldn't bring it up because I think it's a personal style thing, but when mixed in with other code it hurts readability. I think this sort of if statement would be fine if you had a block of them separated away like this: 
```
lstScriptActions.Items.Add(CreateScriptCommandListViewItem(item.ScriptCommand));

if (item.AdditionalScriptCommands.Count > 0) PopulateExecutionCommands(item.AdditionalScriptCommands);
if (item.AdditionalScriptCommands.Count > 0) PopulateSomethingElse(item.AdditionalSomethingElse);
if (item.AdditionalScriptCommands.Count > 0) PopulateSomethingElse2(item.AdditionalSomethingElse2);
...etc
```

But when mixed together with normally indented code it looks like an if statement that has a misplaced parenthesis or a missing logical operator. Since you only have one what I would suggest is doing something like 
```
lstScriptActions.Items.Add(CreateScriptCommandListViewItem(item.ScriptCommand));

if (item.AdditionalScriptCommands.Count > 0) { PopulateExecutionCommands(item.AdditionalScriptCommands); }
```
Or having a new line between the if and the command like we have in other parts of the code base:
```
lstScriptActions.Items.Add(CreateScriptCommandListViewItem(item.ScriptCommand));

if (item.AdditionalScriptCommands.Count > 0)
    PopulateExecutionCommands(item.AdditionalScriptCommands);
```

Reply:
Fixed it by just adding a line in between. I think that's the cleanest option